extensions/visualizations/oppia-visualization-click-hexbins.directive.ts

```diff
@@ -0,0 +1,158 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import * as d3 from 'd3';
+import * as d3Hexbin from 'd3-hexbin';
+
+interface IHexBin {
+  x: number; // X-coordinate of the bin.
+  y: number; // Y-coordinate of the bin.
+  length: number; // Number of clicks grouped into this bin.
+}
+
+/**
+ * @fileoverview Visualization for click statistics providing a hexbin-heatmap.
+ */
+
+angular.module('oppia').directive('oppiaVisualizationClickHexbins', [
```

Comment:
Please change it to an AngularJS component, you can check some examples in this PR: #9317 and #9335

Reply:
Hmm but that'd break the convention of the other visualizations. I'd prefer doing the conversion in a joint PR to keep the consistency in tact.